src/schema/v2/viewingRooms/mutations/updateViewingRoomMutation.ts

```diff
@@ -0,0 +1,72 @@
+import { GraphQLNonNull, GraphQLString } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import { ViewingRoomInputAttributesType } from "./viewingRoomInputAttributes"
+import { ViewingRoomOrErrorType } from "./viewingRoomOrError"
+import { ARImageInputType } from "./ARImageInput"
+import { identity, pickBy } from "lodash"
+
+export const updateViewingRoomMutation = mutationWithClientMutationId<
+  any,
+  any,
+  ResolverContext
+>({
+  name: "updateViewingRoom",
+  inputFields: {
+    attributes: {
+      type: ViewingRoomInputAttributesType,
+    },
+    image: {
+      type: ARImageInputType,
+    },
+    viewingRoomID: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+  },
+  outputFields: {
+    viewingRoomOrErrors: {
+      type: ViewingRoomOrErrorType,
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: async (args, { updateViewingRoomLoader }) => {
+    if (!updateViewingRoomLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    try {
+      // remove undefined or null values from attributes
```

Comment:
Shouldn't we pass null values to Gravity to allow unsetting them? Even if we don't allow null attributes at the moment, we might want to add them in the future.

Reply:
thanks, I will doublecheck, but I think you are right

Reply:
Are undefined values passed on to Gravity? Maybe there is no need to remove them as well 🤔 But I'm not sure about it.

Reply:
Just checked - seems like undefined values are not passed to gravity, but null values are - and this is what we want in this case. I've made a fix, thanks again @olerichter00